Notes.md

```diff
@@ -0,0 +1,38 @@
+https://youtu.be/6K8mW0k7sfw
+
+
+in the guided demo in the knexfile we have both testing and development.both are pointing to different databases its because in testing we need to wipe and manipulate the data as needed, where as in development we want to preserve the data... so needs to be separate 
+
+
+in package .json.... "test":"cross-env DB-env=testing jest--watch--verbose"
+so that testing should connect to the correct database
+knex>knex init>automatically knexfile
+kinex migrate:make createProjectTable,
+knex migrate:latest
+knex seed:make 01-project
+knex seed :run
+
+
+MOST IMP
+when we are running jest with node in package.json we must have
+"jest": {
+		"testEnvironment": "node"
+	},
+    now, jest will not look into the window object.settimeout is node not window.
+
+    we need to have a serverspec.js file to test
+
+    DIFFERENCE BETWEEN TOBE AND TOEQUAL is that toBe works with all primitive data types only int, string not with array and objects for array and objects we need to use toEqual for testing.to Equal  checks just key , value pair
```

Comment:
Yes, `toBe` will check the actual _reference_ of an array or object. That means it has to be, literally, the same object or array. (You'll learn more about this in the computer science curriculum if you don't already know!)